### PR TITLE
feat: [IOPID-1540] Adopt the new DS to `IdentificationModal`

### PR DESCRIPTION
**Pull request by shadowsheep1**
## Short description
This PR addresses the redesign of the `IdentificationModal` by adopting the new DS. 

> [!TIP]
> Review Privacy ✅
> [MP event](https://pagopa.atlassian.net/wiki/spaces/CM1/pages/873758721/TPR+Identificazione+PIN+BIO) ✅
> Translations ✅
>
> [Run e2e tests](https://github.com/pagopa/io-app/actions/workflows/test-e2e.yml)

> [!Warning]
> The little lock icon (🔒) in the `IdentificationModal` layout is the `DevMode` button for fast pin inserting. It's not shown on production builds.

## List of changes proposed in this pull request
- Adopt of the new DS to `IdentificationModal`
- Adopt of the new DS to `IdentificatioLocknModal`
- Make use of a new persisted field in `IdentificationFailData`, for which there is a new `redux-persist` migration with its corresponding unit tests.
- Add CMX properties to `IDENTIFICATION_SUCCESS` event according to [this](https://pagopa.atlassian.net/wiki/spaces/CM1/pages/873758721/TPR+Identificazione+PIN+BIO#%F0%9F%94%84-Eventi-gi%C3%A0-tracciati-da-modificare).
- Make use of `useAppBackgroundAccent` to get the blue color of the DS.
- `ToastNotification` is sticky in the layout because `<Modal>`s will cover the `IOToast` one. `<Modal>`s cover everything that is not inside their layout.

<details><summary>🎦 Demo 🎦</summary>
<p>

| Android a11y  | iOS a11y  | 
| - | - | 
| <video src="https://github.com/pagopa/io-app/assets/16268789/d84c87dc-a5c9-464d-85b6-a06ca9317042" /> | <video src="https://github.com/pagopa/io-app/assets/16268789/3a4239bf-c997-4e48-ab8f-513cc25902ca" /> |

| Android `auth`  | iOS `auth`  | 
| - | - | 
| <video src="https://github.com/pagopa/io-app/assets/16268789/990d4c4d-14d7-4815-aa5e-ef6cb2a2807f" /> | <video src="https://github.com/pagopa/io-app/assets/16268789/50ac4a38-ea56-4547-9e36-c13f37ce2796" /> |

| Android `isValidatingTask`  | iOS `isValidatingTask`  | 
| - | - | 
| <video src="https://github.com/pagopa/io-app/assets/16268789/7169e699-796b-4005-86bd-9532d316b8f7" /> | <video src="https://github.com/pagopa/io-app/assets/16268789/691ad548-b8ca-477e-915a-32b6e1260a06" /> |

| Android `FL`  | iOS `FL`  | 
| - | - | 
| <video src="https://github.com/pagopa/io-app/assets/16268789/e01d7378-d312-4a43-9722-57ef29a3541a" /> | <video src="https://github.com/pagopa/io-app/assets/16268789/5786e9f3-ffc4-4876-9a18-f64f4803b96a" /> |

</p>

</details> 

## How to test
- Test that the authentication with the new modal works frictionless on physical devices with biometric (FaceID, Fingerprint) and no biometric at all.
- Test for error attempts
  - Up to 3 attempts you should see only a "retry" pinned toast notification.
  - At the 4th error you should only see "You have 4 attempts remaining"
  - From the 5th onward to 7th you should see the `IdentificationLockModal` with the corresponding (`30`, `60`, `90`) seconds of waiting. 
    - Restarting the application inside the time gap frame it'll show the lock modal with the remaining time.
    - Restarting the application outside the time gap frame it'll show the auth modal.  
  - With the 8th failure you'll be logged out.
- Test a11y on both Android and iOS devices.
- Check the [MP event](https://pagopa.atlassian.net/wiki/spaces/CM1/pages/873758721/TPR+Identificazione+PIN+BIO).
- Test the modal also on:
  - FL session refresh.
  - Background lock screen after 120 secs.
  - Task validation like (io-sign).

### CODE REVIEW

File: ts/components/core/variables/IOStyleVariables.ts
Comment (dmnplb) [inline ts/components/core/variables/IOStyleVariables.ts:1]: Just don't use this file as a reference. It should be deleted, and it's not a good way to deal with DS-related issues.

**Comment by shadowsheep1** [comment]
@mariagraziapilera copy review needed 🙏

File: ts/utils/identification/index.tsx [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+import { View } from "react-native";
+import I18n from "../../i18n";
+
+export const FAIL_ATTEMPTS_TO_SHOW_ALERT = 4;
+
+export const getBiometryIconName = (
+  biometryPrintableSimpleType: BiometricsValidType
+) => {
+  switch (biometryPrintableSimpleType) {
+    case "BIOMETRICS":
```
Comment (Ladirico) [inline ts/utils/identification/index.tsx:16]: is this case missing the `return`? or is it intentionally so?

File: ts/utils/identification/index.tsx [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+export const getAccessibiliyIdentificationInstructions = (
+  biometricType: BiometricsValidType | undefined,
+  isBimoetricIdentificatoinFailed: boolean = false
+) => {
+  if (isBimoetricIdentificatoinFailed) {
+    return unlockCodeInstruction;
+  }
+
+  switch (biometricType) {
+    case "BIOMETRICS":
```
Comment (Ladirico) [inline ts/utils/identification/index.tsx:49]: as above

File: ts/utils/identification/index.tsx [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+      <LabelSmall color="white"> {faceId}</LabelSmall>
+    </View>
+  );
+
+  if (isBimoetricIdentificatoinFailed) {
+    return instructionComponent;
+  }
+
+  switch (biometricType) {
+    case "BIOMETRICS":
```
Comment (Ladirico) [inline ts/utils/identification/index.tsx:98]: as above

File: ts/utils/identification/index.tsx (hunk 2 above)
Comment (shadowsheep1) [reply ts/utils/identification/index.tsx:49]: As above.

File: ts/utils/identification/index.tsx (hunk 3 above)
Comment (shadowsheep1) [reply ts/utils/identification/index.tsx:98]: As above.

**Review by Ladirico — APPROVED** [review]
LGTM!

**Comment by mariateresaventura** [comment]
> @mariateresaventura copy review needed 🙏

**Comment by mariateresaventura** [comment]
> @mariateresaventura copy review needed 🙏

Done!

**Review by LazyAfternoons — APPROVED** [review]
LGTM for FCI.

**Review by CrisTofani — APPROVED** [review]
LGTM

**Pull request merged**